src/adapters/mocha.js

```diff
@@ -53,12 +53,18 @@ function TesteeReporter(runner) {
       } else if(data.title === '"before all" hook') {
         // tests in this suite will never run if before() fails,
         //  so create the first test in order to fail it.
-        data = data.parent.tests[0];
+        var test;
+        data.parent.eachTest(function(t) {
+          test = test || t;
+        })
+        data = test;
```

Comment:
I think we should preserve the original data if no test is found rather than setting undefined.

Reply:
The way mocha works, if there's no test in the suite or any subsuite, the hook doesn't run (so it shouldn't be able to fail without a test to be found).  But it's not a big deal to make the change, so sure.